source/slang/slang-check-decl.cpp

```diff
@@ -7570,6 +7577,27 @@ namespace Slang
                 }
             }
         }
+        else if (isTypePack(paramDecl->type.type))
+        {
+            // A parameter pack is always `const` if it is not `inout`.
+            bool hasConstModifier = false;
+            for (auto modifier : paramDecl->modifiers)
+            {
+                if (as<OutModifier>(modifier) || as<InOutModifier>(modifier) || as<RefModifier>(modifier) || as<ConstRefModifier>(modifier))
+                {
+                    getSink()->diagnose(modifier, Diagnostics::parameterPackMustBeConst);
+                }
```

Comment:
From what I understand, the code below here is saying is that "A parameter pack is always `const`, otherwise we should error":
`if (as<OutModifier>(modifier) || as<InOutModifier>(modifier) || as<RefModifier>(modifier) || as<ConstRefModifier>(modifier))`

This confuses me since the comment above says, "A parameter pack is always `const` if it is not `inout`". Am I misunderstanding the intention of the code here?

Reply:
I didn't implement `inout` yet, so let me change the comment to match the implementation.